Content.Server/GameObjects/Components/Buckle/BuckleComponent.cs

```diff
@@ -427,15 +430,33 @@ public override ComponentState GetComponentState()
             {
                 drawDepth = BuckledTo.SpriteComponent.DrawDepth - 1;
             }
+            
 
-            return new BuckleComponentState(Buckled, drawDepth);
+            return new BuckleComponentState(Buckled, drawDepth, EntityBuckledTo, DontCollide);
         }
 
         bool IInteractHand.InteractHand(InteractHandEventArgs eventArgs)
         {
             return TryUnbuckle(eventArgs.User);
         }
 
+
+        public void Update()
+        {
+            if (!DontCollide || Body == null)
+                return;
+
+            Body.WakeBody();
```

Comment:
You shouldn't spam WakeBody like this if the body ain't moving because it's expensive for physics.

Reply:
Same as below, took inspiration from the ClimbingComponent.

I need to check if I'm still colliding with the strap entity even when the player is still, otherwise after being unbuckled, the player will be thrown out of the bed if it's not moving.